sympy/core/sympify.py

```diff
@@ -313,11 +324,26 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
             pass
 
     if not isinstance(a, string_types):
-        for coerce in (float, int):
-            try:
-                return sympify(coerce(a))
-            except (TypeError, ValueError, AttributeError, SympifyError):
-                continue
+        if _is_numpy_instance(a):
+            import numpy as np
+            if isinstance(a, np.number):
+                return sympify(a)
+            elif isinstance(a, np.ndarray):
+                # Scalar arrays (those with zero dimensions) have sympify
+                # called on the scalar element.
+                if a.ndim == 0:
+                    try:
+                        return sympify(a.item())
```

Comment:
This should pass the flags through, especially considering the array can be an array of strings.

Reply:
Which is also something we should add tests for.